### PR TITLE
[NEW] Introduce `clearLocalStorageWhenChatEnded` setting logic

### PR DESCRIPTION
**Pull request by tiagoevanp**
Introduce the logic of Omnichannel/Livechat `Livechat_clear_local_storage_when_chat_ended` setting

### CODE REVIEW

File: src/lib/room.js [hunk 1]
```diff
@@ -21,6 +21,13 @@ export const closeChat = async ({ transcriptRequested } = {}) => {
 		await handleTranscript();
 	}
 
+	const { config: { settings: { removeGuestInfoWhenChatEnded } = {} } = {} } = store.state;
+
+	if (removeGuestInfoWhenChatEnded) {
```
Comment (renatobecker) [inline src/lib/room.js:26]: Visitor won't be deleted, just the session will be clean/reset.

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: src/store/Store.js [hunk 2]
```diff
@@ -77,6 +77,10 @@ export default class Store {
 		this.emit('change', [this._state, prevState, partialState]);
 	}
 
+	cleanState() {
+		this._state = {};
```
[the remark this answers is not in the document]
Comment (KevLehman) [reply src/store/Store.js:81]: Hey @renatobecker , can you take a look again?

[the remark this answers is not in the document]
Comment (renatobecker) [reply src/store/Store.js:81]: It works like a charm now, but we should not change the following fields as they impact UI/UX.
- minimized
- visible
- undocked
- expanded

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: src/lib/room.js [hunk 3]
```diff
@@ -21,6 +21,12 @@ export const closeChat = async ({ transcriptRequested } = {}) => {
 		await handleTranscript();
 	}
 
+	const { config: { settings: { clearLocalStorageWhenChatEnded } = {} } = {} } = store.state;
+
+	if (clearLocalStorageWhenChatEnded) {
+		await store.setState(initialState());
```
Comment (renatobecker) [inline src/lib/room.js:27]:
```suggestion
		const { visible, minimized, undocked, expanded, ...newState } = initialState();
		await store.setState(newState);
```

WDYT?

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: src/store/Store.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (KevLehman) [reply src/store/Store.js:81]: Changes done!

File: src/lib/room.js (hunk 3 above)
Comment (KevLehman) [reply src/lib/room.js]: Yup, did basically the same haha

File: src/store/Store.js [hunk 4]
```diff
@@ -77,6 +77,10 @@ export default class Store {
 		this.emit('change', [this._state, prevState, partialState]);
 	}
 
+	cleanState() {
```
Comment (renatobecker) [inline src/store/Store.js:80]: This change is no longer needed.

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: src/store/Store.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (renatobecker) [reply src/store/Store.js:81]: LGTM, but please check [this comment](https://github.com/RocketChat/Rocket.Chat.Livechat/pull/666#discussion_r7613774990)

File: src/store/Store.js (hunk 4 above)
Comment (KevLehman) [reply src/store/Store.js]: Removed

**Comment by KevLehman** [comment]
After store change I had to update storybook, that generated a ton more of changes :(

**Review by renatobecker — APPROVED** [review]

**Pull request merged**